### PR TITLE
Initial search/replace of floxdev.com->flox.dev

### PR DESCRIPTION
**Pull request by garbas**
This PR should be merged once website/discourse/events.flox.dev domain are ready.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,11 +24,11 @@
 
 <h3 align="center">
    &emsp;
-   <a href="https://discourse.floxdev.com"><b>Discourse</b></a>
+   <a href="https://discourse.flox.dev"><b>Discourse</b></a>
```
Comment (tomberek) [inline README.md:27]: Let's double-check that discourse will move.

File: README.md [hunk 2]
```diff
@@ -111,15 +111,15 @@ guide](./CONTRIBUTING.md) first.
 The flox CLI is licensed under the GPLv2. See [LICENSE](./LICENSE).
 
 
-[website]: https://floxdev.com
-[discourse]: https://discourse.floxdev.com
+[website]: https://flox.dev
+[discourse]: https://discourse.flox.dev
```
Comment (tomberek) [inline README.md:115]: double-check

File: doc/release-notes/rl-0.1.4.md [hunk 3]
```diff
@@ -13,4 +13,4 @@ This release includes several performance improvements, bug fixes, and feature r
 - Updated output level of internal commands to forward more prompts and warnings to users, e.g. to provide passwords or yubikey input.
 - Squashed some bugs, improved error messages / UX output, updated manpages and fixed typos.
 
-We especially want to thank all our <a href="https://github.com/flox/flox">github.com/flox/flox</a> contributors and <a href="https://discourse.floxdev.com">discourse community</a> members for all your valuable feedback!
+We especially want to thank all our <a href="https://github.com/flox/flox">github.com/flox/flox</a> contributors and <a href="https://discourse.flox.dev">discourse community</a> members for all your valuable feedback!
```
Comment (tomberek) [inline doc/release-notes/rl-0.1.4.md:16]: ditto

File: README.md (hunk 1 above)
Comment (garbas) [reply README.md:27]: Didn't we say we are migrating to `flox.dev` domain? That means everything including discourse, right? If not, who makes the decision so I can talk to?

Comment (garbas) [reply README.md:27]: discourse.flox.dev is now live

File: README.md (hunk 2 above)
Comment (garbas) [reply README.md:115]: discourse.flox.dev is now live

File: doc/release-notes/rl-0.1.4.md (hunk 3 above)
Comment (garbas) [reply doc/release-notes/rl-0.1.4.md:16]: I think it is fine if we change them. no harm.

**Review by mkenigs — APPROVED** [review]

**Review by rossturk — APPROVED** [review]

**Pull request merged**